mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoScaler.java

```diff
@@ -271,80 +273,25 @@ protected Map<Integer, ClutchConfiguration> getClutchConfiguration(String jsonCo
         }).get());
     }
 
+    @Value
+    @RequiredArgsConstructor
     public static class Event {
 
-      private final StageScalingPolicy.ScalingReason type;
-      private final int stage;
-      private final double value;
-      private final int numWorkers;
-      private final String message;
-
-      public Event(StageScalingPolicy.ScalingReason type, int stage, double value, int numWorkers, String message) {
-        this.type = type;
-        this.stage = stage;
-        this.value = value;
-        this.numWorkers = numWorkers;
-        this.message = message;
-      }
-
-      public StageScalingPolicy.ScalingReason getType() {
-        return type;
-      }
-
-      public int getStage() {
-        return stage;
-      }
-
-      public double getValue() {
-        return value;
-      }
-
-      public int getNumWorkers() {
-        return numWorkers;
-      }
-
-      public String getMessage() {
-        return message;
-      }
-
-      @Override
-      public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        Event event = (Event) o;
-
-        if (stage != event.stage) return false;
-        if (Double.compare(event.value, value) != 0) return false;
-        if (numWorkers != event.numWorkers) return false;
-        if (type != event.type) return false;
-        return message != null ? message.equals(event.message) : event.message == null;
-
-      }
-
-      @Override
-      public int hashCode() {
-        int result;
-        long temp;
-        result = type != null ? type.hashCode() : 0;
-        result = 31 * result + stage;
-        temp = Double.doubleToLongBits(value);
-        result = 31 * result + (int) (temp ^ (temp >>> 32));
-        result = 31 * result + numWorkers;
-        result = 31 * result + (message != null ? message.hashCode() : 0);
-        return result;
-      }
-
-      @Override
-      public String toString() {
-        return "Event{" +
-          "type=" + type +
-          ", stage=" + stage +
-          ", value=" + value +
-          ", numWorkers=" + numWorkers +
-          ", message='" + message + '\'' +
-          '}';
-      }
+      StageScalingPolicy.ScalingReason type;
+      int stage;
+      double value;
+      double effectiveValue;
+      int numWorkers;
+      String message;
+
+        public Event(ScalingReason type, int stage, double value, double effectiveValue, int numWorkers) {
+            this.type = type;
+            this.stage = stage;
+            this.value = value;
```

Comment:
do we still need value? why not just effectiveValue?

Reply:
It is still being used in some places.